File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, w1: str, w2: str) -> str:
        return ''.join(a+b for a,b in zip_longest(w1,w2,fillvalue=''))
        
# class Solution:
#     def mergeAlternately(self, word1: str, word2: str) -> str:
#         n,m,i = len(word1),len(word2),0
#         res = ""
#         while(i<n and i<m):
#             res += word1[i]+word2[i]
#             i+=1
#         if(i==n):
#             res += word2[i:]
#         else:
#             res += word1[i:]
#         return res

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string w1, string w2) {
        string res = "";
        for (int i = 0; i < w1.size() || i < w2.size(); ++i) {
            if (i < w1.size())
                res.push_back(w1[i]);
            if (i < w2.size())
                res.push_back(w2[i]);
        }
        return res;
    }
};

// class Solution {
// public:
//     string mergeAlternately(string w1, string w2) {
//         int n = w1.size(), m = w2.size(), i = 0, j = 0;
//         string res = "";
//         while (i < n || j < m) {
//             if (i < n)
//                 res.push_back(w1[i++]);
//             if (j < m)
//                 res.push_back(w2[j++]);
//         }
//         return res;
//     }
// };